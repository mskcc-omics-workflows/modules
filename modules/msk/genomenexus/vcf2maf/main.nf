process GENOMENEXUS_VCF2MAF {
    tag "$meta.id"
    label 'process_low'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'ghcr.io/msk-access/genomenexus:vcf2maf_lite':
        'ghcr.io/msk-access/genomenexus:vcf2maf_lite' }"

    input:
    tuple val(meta), path(vcf)

    output:
    tuple val(meta), path("vcf2maf_output/${meta.control_id}_${meta.case_id}_annotated.maf"), emit: maf
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    python3 /vcf2maf-lite/vcf2maf_lite.py -i ${vcf} ${args}
    echo '"${task.process}": vcf2maf_lite.py v.0.0.1' > versions.yml
    """
    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"

    """

    mkdir vcf2maf_output
    touch ${meta.control_id}_${meta.case_id}_annotated.maf
    cp ${meta.control_id}_${meta.case_id}_annotated.maf vcf2maf_output/
    echo '"${task.process}": vcf2maf_lite.py v.0.0.1' > versions.yml
    """
}
