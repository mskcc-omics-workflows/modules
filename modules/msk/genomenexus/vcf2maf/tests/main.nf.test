
// nf-core modules test genomenexus/vcf2maf
nextflow_process {

    name "Test Process GENOMENEXUS_VCF2MAF"
    script "../main.nf"
    process "GENOMENEXUS_VCF2MAF"
    

    tag "modules"
    tag "modules_nfcore"
    tag "modules_msk"
    tag "genomenexus"
    tag "genomenexus/vcf2maf"

    
    test("vcf2maf - test") {

        when {
            process {
                """
                
                
                input[0] = [
                    [ id:'test'], // meta map
                    file(params.test_data['sarscov2']['illumina']['test_vcf'], checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success }
                
                
            )
        }

    }

    
    test("vcf2maf - test - stub") {

        options "-stub"

        when {
            process {
                """
               
                
                input[0] = [
                    [ id:'test'], // meta map
                    file(params.test_data['sarscov2']['illumina']['test_vcf'], checkIfExists: true)
                    ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                
            )
        }

    }

}
