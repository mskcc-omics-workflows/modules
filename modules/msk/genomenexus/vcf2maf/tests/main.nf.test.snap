{
    "vcf2maf - test - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.maf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,c60e8688d2d8110c4621c279a710a890"
                ],
                "maf": [
                    [
                        {
                            "id": "test"
                        },
                        "test.maf:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,c60e8688d2d8110c4621c279a710a890"
                ]
            }
        ],
        "timestamp": "2024-06-24T11:56:07.276674"
    }
}